File: CodeWarriors.IITDU/CodeWarriors.IITDU/Controllers/ProfileController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using CodeWarriors.IITDU.Models;
using CodeWarriors.IITDU.Service;
using CodeWarriors.IITDU.ViewModels;

namespace CodeWarriors.IITDU.Controllers
{
    [Authorize]
    public class ProfileController : Controller
    {
        private AccountService _accountService;
        private ProfileModel _profileModel;
        private UserProfile _profile;
        public ProfileController(AccountService accountService, ProfileModel profileModel, UserProfile profile)
        {
            _accountService = accountService;
            _profileModel = profileModel;
            _profile = profile;
        }
        public ActionResult Index()
        {
            var profile = _accountService.GetProfile(User.Identity.Name);
            _profileModel.FirstName = profile.FirstName;
            _profileModel.LastName = profile.LastName;
            _profileModel.About = profile.About;
            _profileModel.Age = profile.Age;
            _profileModel.Gender = profile.Gender;
            _profileModel.Location = profile.Location;

            _profileModel.ImageUrl = profile.ImageUrl;
            return View(_profileModel);
        }

        [HttpGet]
        public JsonResult GetProfileInformation()
        {
            var profileInformation = _accountService.GetProfile(User.Identity.Name);
            return Json(profileInformation, JsonRequestBehavior.AllowGet);
        }

        [AllowAnonymous]
        [HttpGet]
        public JsonResult GetProfileInformationByUserId(int userId)
        {
            var profileInformation = _accountService.GetUserProfileByUserId(userId);
            return Json(profileInformation, JsonRequestBehavior.AllowGet);
        }

        [HttpPost]
        public JsonResult UpdateProfile(ProfileModel model)
        {
            if (ModelState.IsValid)
            {
                _profile.FirstName = model.FirstName;
                _profile.LastName = model.LastName;
                _profile.Age = model.Age;
                _profile.About = model.About;
                _profile.Location = model.Location;
                _profile.MobileNumber = model.MobileNumber;
                _profile.Gender = model.Gender;
                if (model.ImageUrl.Contains("Upload/"))
                    _profile.ImageUrl = model.ImageUrl;
                else
                    _profile.ImageUrl = "Upload/" + model.ImageUrl;
                _accountService.EditProfile(_profile, User.Identity.Name);
                return Json("Updated Successfully", JsonRequestBehavior.AllowGet);
            }
            return Json("Sorry, Could Not Update", JsonRequestBehavior.AllowGet);
        }

        [HttpPost]
        public JsonResult UpdateProfilePicture(ProfileModel model)
        {
            _profile.FirstName = model.FirstName;
            _profile.LastName = model.LastName;
            _profile.Age = model.Age;
            _profile.About = model.About;
            _profile.Location = model.Location;
            _profile.MobileNumber = model.MobileNumber;
            _profile.Gender = model.Gender;
            if (model.ImageUrl.Contains("Upload/"))
                _profile.ImageUrl = model.ImageUrl;
            else
                _profile.ImageUrl = "Upload/" + model.ImageUrl;
            _accountService.EditProfile(_profile, User.Identity.Name);
            return Json("Updated Successfully", JsonRequestBehavior.AllowGet);
        }

        public ActionResult Edit()
        {
            var profile = _accountService.GetProfile(User.Identity.Name);
            _profileModel.FirstName = profile.FirstName;
            _profileModel.LastName = profile.LastName;
            _profileModel.About = profile.About;
            _profileModel.Age = profile.Age;
            _profileModel.Gender = profile.Gender;
            _profileModel.Location = profile.Location;
            _profileModel.MobileNumber = profile.MobileNumber;
            _profileModel.ImageUrl = profile.ImageUrl;
            return View(_profileModel);
        }
        [HttpPost]
        public ActionResult Edit(ProfileModel model)
        {
            if (ModelState.IsValid)
            {
                _profile.FirstName = model.FirstName;
                _profile.LastName = model.LastName;
                _profile.Age = model.Age;
                _profile.About = model.About;
                _profile.Location = model.Location;
                _profile.MobileNumber = model.MobileNumber;
                _profile.Gender = model.Gender;
                _accountService.EditProfile(_profile, User.Identity.Name);
                return RedirectToAction("Index");
            }
            return View(model);
        }
    }
}

File: CodeWarriors.IITDU/CodeWarriors.IITDU/App/Controllers/ProfileController.js
﻿app.controller("ProfileController", function ($scope, $http, uploadManager, $rootScope, $location) {
    init();
    function init() {
        $http.get("/Account/IsAuthenticated").success(function (response) {
            if (!response) {
                $location.path("home");
            }
        });
    }
    $scope.EditMode = false;
    $scope.ProfileInfo = {};
    $scope.NewProfileInfo = {};
    $scope.MapSource = "";
    $scope.Message = "";
    $http.get("/Profile/GetProfileInformation").success(function (response) {
        $scope.ProfileInfo = response;
        if ($scope.ProfileInfo.ImageUrl == null) {
            $scope.ProfileInfo.ImageUrl = "Upload//default.jpg";
        }
        $scope.MapSource = "/Map/ShowMap?location=" + $scope.ProfileInfo.Location;
    });

    $scope.EditProfile = function () {
        $scope.EditMode = true;
        $scope.CopyProfileObject($scope.NewProfileInfo, $scope.ProfileInfo);
        $scope.Message = "";
    }

    $scope.CancelEdit = function () {
        $scope.EditMode = false;
    }

    $scope.ResetInfo = function () {
        $scope.CopyProfileObject($scope.NewProfileInfo, $scope.ProfileInfo);
        //uploadManager.clear();
    }

    $scope.UpdateProfile = function () {
        $http.post("/Profile/UpdateProfile", $scope.NewProfileInfo).success(function (response) {
            $scope.CopyProfileObject($scope.ProfileInfo, $scope.NewProfileInfo);
            $scope.EditMode = false;
            $scope.Message = response;
        });
    }

    $scope.CopyProfileObject = function (a, b) {
        a.FirstName = b.FirstName;
        a.LastName = b.LastName;
        a.Age = b.Age;
        a.About = b.About;
        a.Location = b.Location;
        a.Gender = b.Gender;
        a.MobileNumber = b.MobileNumber;
        a.ImageUrl = b.ImageUrl;
    }


    $scope.files = [];

    $scope.percentage = 0;

    $scope.upload = function () {
        uploadManager.uploadProfile();
    };

    $scope.$on('uploadProfileDone', function (e, call) {
        $scope.CopyProfileObject($scope.NewProfileInfo, $scope.ProfileInfo);
        var allFiles = uploadManager.files();
        $scope.NewProfileInfo.ImageUrl = allFiles[allFiles.length - 1];
        $http.post("/Profile/UpdateProfilePicture", $scope.NewProfileInfo).success(function (response) {
            alert('done');
        });

    });


    $scope.$on('fileAdded', function (e, call) {
        $scope.files = [];
        $scope.files.push(call);
        $scope.$apply();
    });

});